src/main/java/pro/trevor/tankgame/state/attribute/Attributes.java

```diff
@@ -0,0 +1,12 @@
+package pro.trevor.tankgame.state.attribute;
+
+public class Attributes {
+
+    public static GoldAttribute GOLD = new GoldAttribute();
```

Comment:
Could you directly construct base base attribute here with something like `public static GoldAttribute GOLD = new BaseAttribute("GOLD", Integer.class);`?

Reply:
Yeah, I think so

Reply:
I'd be missing out on the reflection (i think that's what it's called) where the type can be shown in the error.